src/pages/Profile/index.jsx

```diff
@@ -72,12 +73,17 @@ function Profile() {
       docRef
         .get()
         .then((snapshot) => {
-          const doc = snapshot.docs[0];
-          setUserData({
-            name: doc.data().name,
-            avatar: doc.data().photoURL,
-            uid: doc.data().uid,
-          });
+          if(snapshot.exists){
```

Comment:
This is perhaps not a valid condition and hence profile was being forwarded to 404 page everytime instead.

Reply:
I had to revert this, sort of, to fix this